java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -339,6 +339,52 @@ private static void dumpConfig(final ConfigContext context) {
         }
     }
 
+
+    /* ======================================================================
+     * MantaClient content-type Utility Methods
+     * ====================================================================== */
+
+    /**
+     * Depending on the configuration value to enable/disable automatic content-type detection
+     * within the Client SDK {@link ConfigContext#isContentTypeDetectionEnabled()}, assign
+     * content-type value accordingly.
+     *
+     * @param headers optional HTTP headers to include when copying the object.
+     * @return content type object
+     */
+    private ContentType assignContentType(final MantaHttpHeaders headers) {
+        if (BooleanUtils.isTrue(config.isContentTypeDetectionEnabled())) {
```

Comment:
I just looked at the code more closely. The method `ContentTypeLookup.findOrDefaultContentType(headers,
 ContentType.APPLICATION_OCTET_STREAM)` doesn't do content-type detection. It just assigns the content-type indicated in the headers object. We don't want to disable that operation. We only want to allow for the disabling of code paths that call `findOrDefaultContentType(final MantaHttpHeaders headers, final String filename, final File file, final ContentType defaultContentType)`.

Reply:
You're right, sorry !

Reply:
Noted and resolved.